File: automations/6999d106f9e774801b921d58/Normalize Inputs.js
import logging
from typing import Dict, Any
import json

logging.basicConfig(level=logging.INFO, format='%(asctime)s [%(levelname)s] %(message)s')

def normalize_inputs(data: Dict[str, Any]) -> Dict[str, Any]:
    """
    Normalizes input data to standard formats and trims whitespace.
    Args:
        data (Dict[str, Any]): Validated intake data.
    Returns:
        Dict[str, Any]: Normalized input data.
    """
    try:
        normalized = {}
        for key, value in data.items():
            if isinstance(value, str):
                normalized[key] = value.strip().lower()
            else:
                normalized[key] = value
        logging.info('Input normalization successful.')
        return normalized
    except Exception as ex:
        logging.error(f'Normalization error: {ex}')
        raise

if __name__ == "__main__":
    import os
    PREV_OUTPUT = os.environ.get('VALIDATED_INTAKE_JSON', '{}')
    try:
        validated_intake = json.loads(PREV_OUTPUT)
    except Exception as e:
        logging.error(f'Failed to load previous step output: {e}')
        raise
    normalized = normalize_inputs(validated_intake)
    print(json.dumps({'normalized_inputs': normalized}))

File: automations/6999d106f9e774801b921d58/Normalize Inputs.py
from runtime import getContext, setContext
import logging
import os
import json
from datetime import datetime, UTC
from typing import Dict, Any

def build_normalized_mepp(data: Dict[str, Any]) -> Dict[str, Any]:
    # Normalize and strictly map to nested MEPP contract
    def extract(first, dct, typ=str, fallback=None):
        v = dct.get(first, fallback)
        return v if typ is None else (v if isinstance(v, typ) else fallback)

    # Only use input where present; do not forcibly lowercase names/addresses, only trim whitespace
    issue = {
        'summary': extract('summary', data, str, '').strip() if extract('summary', data, str, None) else '',
        'details': extract('details', data, str, '').strip() if extract('details', data, str, None) else '',
        'category': extract('category', data, str, '').strip() if extract('category', data, str, None) else '',
    }
    location = {
        'lat': float(data['lat']) if 'lat' in data and data['lat'] is not None else None,
        'lon': float(data['lon']) if 'lon' in data and data['lon'] is not None else None,
        'address_text': extract('address_text', data, str, '').strip() if extract('address_text', data, str, None) else '',
        'ward': extract('ward', data, str, '').strip() if extract('ward', data, str, None) else '',
    }
    contact_type = data.get('contact_type', '').strip() or 'none'
    contact_value = data.get('contact_value', '').strip() if data.get('contact_value') else ''
    reporter = {
        'contact': {'type': contact_type, 'value': contact_value},
        'anonymity_requested': bool(data.get('anonymity_requested', False)),
    }
    norm = {
        'MEPP': {
            'issue': issue,
            'location': location,
            'evidence': {},
            'provenance': {},
        },
        'reporter': reporter
    }
    return norm

if __name__ == "__main__":
    try:
        PREV_OUTPUT = os.environ.get('VALIDATED_INTAKE_JSON', '{}')
        validated_intake = json.loads(PREV_OUTPUT)
        normalized_inputs = build_normalized_mepp(validated_intake.get('validated_intake', validated_intake))
        timestamp = datetime.now(UTC).isoformat()
        setContext('NORMALIZED_INPUTS', {'normalized_inputs': normalized_inputs, 'normalized_at': timestamp})
        print(json.dumps({'normalized_inputs': normalized_inputs, 'normalized_at': timestamp}))
    except Exception as ex:
        logging.error(f'Normalization error: {ex}')
        fallback = {'normalized_inputs': {}, 'normalized_at': datetime.now(UTC).isoformat(), 'error': str(ex)}
        setContext('NORMALIZATION_ERROR', fallback)
        print(json.dumps(fallback))